### PR TITLE
Move param creation from BTPayPalDriver to BTPayPalRequest

### PR DESCRIPTION
**Pull request by sestevens**
### Summary of changes

We're planning to create two subclasses of `BTPayPalRequest` -- one for billing agreement (aka vault) flows and one for one-time payment (aka checkout) flows. This PR sets up us to be able to do that work more easily by moving the logic to create HTTP request params out of `BTPayPalDriver` into `BTPayPalRequest`. Eventually, each subclass of `BTPayPalRequest` will be able to construct its own params, and `BTPayPalDriver` can be fairly agnostic about which type of request it's using.

### Checklist

- ~Added a changelog entry~

- @sestevens

### CODE REVIEW

File: Sources/BraintreePayPal/BTPayPalDriver.m [hunk 1]
```diff
@@ -147,88 +143,10 @@ - (void)requestPayPalCheckout:(BTPayPalRequest *)request
             return;
         }
 
-        NSMutableDictionary *parameters = [NSMutableDictionary dictionary];
```
Comment (sestevens) [inline Sources/BraintreePayPal/BTPayPalDriver.m:150]: This chunk of code for creating the parameters dictionary now lives in `BTPayPalRequest`.

File: Sources/BraintreePayPal/BTPayPalDriver.m [hunk 2]
```diff
@@ -29,15 +28,12 @@
...
 #import <BraintreeCore/BTLogger_Internal.h>
-#import <BraintreePayPal/BTPayPalRequest.h>
 #import <BraintreePayPal/BTConfiguration+PayPal.h>
 #import <BraintreePayPal/BTPayPalLineItem.h>
 #import <PayPalDataCollector/PayPalDataCollector-Swift.h>
 #endif
 
 NSString *const BTPayPalDriverErrorDomain = @"com.braintreepayments.BTPayPalDriverErrorDomain";
-NSString *const BTCallbackURLHostAndPath = @"onetouch/v1/";
-NSString *const BTCallbackURLScheme = @"sdk.ios.braintree";
```
Comment (sestevens) [inline Sources/BraintreePayPal/BTPayPalDriver.m:40]: I moved these two constants to `BTPayPalRequest`.

File: Sources/BraintreePayPal/BTPayPalDriver.m [hunk 3]
```diff
@@ -515,38 +433,6 @@ + (BTPayPalAccountNonce *)payPalAccountFromJSON:(BTJSON *)payPalAccount {
     return tokenizedPayPalAccount;
 }
 
-+ (NSString *)intentTypeToString:(BTPayPalRequestIntent)intentType {
```
Comment (sestevens) [inline Sources/BraintreePayPal/BTPayPalDriver.m:518]: These two methods now live in `BTPayPalRequest`.

**Review by scannillo — APPROVED** [review]
⭐️

File: Sources/BraintreePayPal/BTPayPalDriver.m [hunk 4]
```diff
@@ -29,15 +28,12 @@
...
 #import <BraintreeCore/BTPaymentMethodNonceParser.h>
 #import <BraintreeCore/BTLogger_Internal.h>
-#import <BraintreePayPal/BTPayPalRequest.h>
 #import <BraintreePayPal/BTConfiguration+PayPal.h>
 #import <BraintreePayPal/BTPayPalLineItem.h>
 #import <PayPalDataCollector/PayPalDataCollector-Swift.h>
 #endif
 
 NSString *const BTPayPalDriverErrorDomain = @"com.braintreepayments.BTPayPalDriverErrorDomain";
-NSString *const BTCallbackURLHostAndPath = @"onetouch/v1/";
```
Comment (hollabaq86) [inline Sources/BraintreePayPal/BTPayPalDriver.m:39]: do we still need one touch? My memory is probably horribly off but I thought that's a 💀  integration

File: Sources/BraintreePayPal/BTPayPalRequest.m [hunk 5]
```diff
@@ -32,4 +37,108 @@ - (instancetype)initWithAmount:(NSString *)amount {
...
+    if (self.localeCode) {
+        experienceProfile[@"locale_code"] = self.localeCode;
+    }
+
+    if (self.merchantAccountID) {
+        parameters[@"merchant_account_id"] = self.merchantAccountID;
+    }
+
+    // Currency code should only be used for Hermes Checkout (one-time payment).
+    // For BA, currency should not be used.
```
Comment (hollabaq86) [inline Sources/BraintreePayPal/BTPayPalRequest.m:74]:
```suggestion
    // For Billing Agreements (vault), currency should not be used.
```

**Review by hollabaq86 — APPROVED** [review]
Love this refactor. My comments are super small and general Qs about the state of PP integrations in Native (my memory is rusty!)

**Review by sshropshire — APPROVED** [review]

File: Sources/BraintreePayPal/BTPayPalDriver.m (hunk 4 above)
Comment (sestevens) [reply Sources/BraintreePayPal/BTPayPalDriver.m:39]: [One Touch](https://www.paypal.com/us/webapps/mpp/one-touch-checkout) is still a thing. You're probably thinking of the code from the OneTouch SDK that was added into our SDK a long time ago and which we recently deleted.

**Pull request merged**